src/jvmTest/kotlin/com/charleskorn/kaml/JvmYamlWritingTest.kt

```diff
@@ -176,6 +178,121 @@ class JvmYamlWritingTest : DescribeSpec({
+            val output = ByteArrayOutputStream()
+            val thing =  ThingSL(
+                "Name of Thing",
+                "String 1\nString 2\nString 3\n",
+                "Literal 1\nLiteral 2\nLiteral 3\n",
+                "Folded 1\nFolded 2\nFolded 3\n",
+                "Plain 1\nPlain 2\nPlain 3\n",
+            )
+
+            Yaml.default.encodeToStream<ThingSL>(thing, output)
+            output.toString(Charsets.UTF_8) shouldBe
+                """
+                    name: "Name of Thing"
+                    string: "String 1\nString 2\nString 3\n"
+                    literal: "Literal 1\nLiteral 2\nLiteral 3\n"
+                    folded: "Folded 1\nFolded 2\nFolded 3\n"
+                    plain: "Plain 1\nPlain 2\nPlain 3\n"
+                    
+                """.trimIndent()
+        }
+
+        describe("serializing a string as an explicitly stated ScalarStyle (Single Line - ThingML)") {
+            val output = ByteArrayOutputStream()
+            val thing =  ThingML(
+                "Name of Thing",
+                "String",
+                "Literal",
+                "Folded",
+                "Plain",
+            )
+
+            Yaml.default.encodeToStream<ThingML>(thing, output)
+            output.toString(Charsets.UTF_8) shouldBe
+                """
+                    name: "Name of Thing"
+                    string: "String"
+                    literal: "Literal"
+                    folded: "Folded"
+                    plain: "Plain"
+
+                """.trimIndent()
+        }
+
+        describe("serializing a string as an explicitly stated ScalarStyle (Multi Line - ThingML)") {
+            val output = ByteArrayOutputStream()
+            val thing =  ThingML(
+                "Name of Thing",
+                "String 1\nString 2\nString 3\n",
+                "Literal 1\nLiteral 2\nLiteral 3\n",
+                "Folded 1\nFolded 2\nFolded 3\n",
+                "Plain 1\nPlain 2\nPlain 3\n",
+            )
+
+            Yaml.default.encodeToStream<ThingML>(thing, output)
+            output.toString(Charsets.UTF_8) shouldBe
+                """
+                    name: "Name of Thing"
+                    string: "String 1\nString 2\nString 3\n"
+                    literal: |
+                      Literal 1
+                      Literal 2
+                      Literal 3
+                    folded: >
+                      Folded 1
+                    
+                      Folded 2
+                    
+                      Folded 3
+                    plain: 'Plain 1
+                    
+                      Plain 2
+                    
+                      Plain 3
+                    
+                      '
+                
+                """.trimIndent()
+        }
+
+        describe("Ensure all StringScalarStyle and ScalarStyle values match 1 on 1") {
+            // Mapping the StringScalarStyles to ScalarStyle must result in 100% coverage
```

Comment:
I don't think this test is necessary. There are a fixed set of styles in YAML, and we have support for all of them already.

Reply:
Removed